authentication/views.py

```diff
@@ -77,3 +93,40 @@ def post(self, request):
             print("Exception", e)
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
+
+class PasswordRecoveryAPIView(APIView):
+    def post(self, request):
+        email = request.data.get('email')
+        try:
+            validate_email(email)
+        except ValidationError:
+            return Response({'error': 'Invalid email address'}, status=400)
+
+        try:
+            user = CustomUser.objects.get(email=email)
+        except CustomUser.DoesNotExist:
+            return Response({'error': 'User with this email does not exist'}, status=404)
+
+        jwt_token = Utils.generate_token(user.email, user.pk)
+        reset_link = f"{settings.FRONTEND_URL}/auth/password-reset/{jwt_token}/"
+        try:
+            Utils.send_password_reset_email(email, reset_link)
+        except Exception as e:
+            return Response({'error': 'Failed to send email'}, status=500)
+        return Response({'message': 'Password reset email sent successfully'}, status=status.HTTP_200_OK)
+
+
+
+class PasswordResetView(APIView):
+    serializer_class = PasswordRecoverySerializer
+
+    def post(self, request, jwt_token):
+        uid, email, exp = Utils.decode_token(jwt_token)
+        user = Utils.get_user(uid, email)
```

Comment:
Ensure that ```decode_token``` and ```get_user methods``` handle all potential edge cases, such as expired tokens or users that no longer exist. It seems you return ```None``` values from ```decode_token``` in case of exceptions, which is a good practice, but ensure you check these ```None``` values before proceeding to ```get_user```